frontend/src/__tests__/Stats.test.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import wait from 'waait'
+import { mount } from 'enzyme'
+import { MockedProvider } from 'react-apollo/test-utils'
+import { I18nProvider } from '@lingui/react'
+import gql from 'graphql-tag'
+import { Stats } from '../Stats'
+import en from '../../locale/en/messages.js'
+const catalogs = { en }
+
+describe('<Stats/>', () => {
+  describe('when the API returns data', () => {
+    it('displays stats', async () => {
+      let mocks = [
+        {
+          request: {
+            query: gql`
+              query GetStats {
+                stats {
+                  reportCount
+                }
+              }
+            `,
+          },
+          result: { data: { stats: { reportCount: 5 } } },
+        },
+      ]
+
+      let wrapper = mount(
+        <MockedProvider mocks={mocks} addTypename={false}>
+          <I18nProvider language={'en'} catalogs={catalogs}>
+            <Stats />
+          </I18nProvider>
+        </MockedProvider>,
+      )
+
+      await wait(0)
+      expect(wrapper.text()).toMatch(/You are the 5th person/)
+    })
+  })
+
+  describe('when loading', async () => {
+    it('displays an empty string', async () => {
+      let mocks = [
+        {
+          request: {
+            query: gql`
+              query GetStats {
+                stats {
+                  reportCount
+                }
+              }
+            `,
+          },
+          result: { data: { stats: { reportCount: 5 } } },
+        },
+      ]
+
+      let wrapper = mount(
+        <MockedProvider mocks={mocks} addTypename={false}>
+          <I18nProvider language={'en'} catalogs={catalogs}>
+            <Stats />
+          </I18nProvider>
+        </MockedProvider>,
+      )
+
+      expect(wrapper.text()).toMatch('')
+    })
+  })
+
+  describe('when an error is raised', async () => {
+    it('displays an error message', async () => {
+      let mocks = [
+        {
+          request: {
+            query: gql`
+              query GetStats {
+                stats {
+                  reportCount
+                }
+              }
+            `,
+          },
+          error: new Error('sadness'),
+        },
+      ]
+
+      let wrapper = mount(
+        <MockedProvider mocks={mocks} addTypename={false}>
+          <I18nProvider language={'en'} catalogs={catalogs}>
+            <Stats />
+          </I18nProvider>
+        </MockedProvider>,
+      )
+
+      await wait(0)
```

Comment:
wait(0)?

Reply:
If you throw some console.log statements into the "function as a child" that we are passing to the Query component you can see it's being called multiple times. This lets the promise that the client returns resolve so our Query component code gets called again with the actual data.